HW0/P5/P5.txt

```diff
@@ -1 +1,33 @@
 # Your answers here.
+1. 255 s
+   2^7 + 2^6 + ... + 2^0
+
+2. Yes, faster. 34 s
+   2^7/(2^3) + 2^6/(2^3) + ...+ 2^3/(2^3) 
+   + 2^2/(2^3)[==1] + 2^1/(2^3)[==1] + 2^0/(2^3)[==1]
+
+3. 8 s
+   2^7/(inf)[==1] + 2^6/(inf)[==1] + ...+ 2^3/(inf)[==1] 
+   + 2^2/(inf)[==1] + 2^1/(inf)[==1] + 2^0/(inf)[==1]
+
+4. log2(N)
+
+5. 15 s
+   7(communication) + 8(counting)
+
+6. 1 bag case: 272s
```

Comment:
For both 6 and 7, I agree with your logic, but do you consider the scenario where communication between cashiers can actually occur in parallel to the handing out? (ie, give 128 bags to the first 128 cashiers, and by the time you give out the next 128 bags, the first 128 cashiers will have already finished summing their bag's values?)